xbmc/music/Album.cpp

```diff
@@ -206,12 +206,58 @@ CAlbum::CAlbum(const CFileItem& item)
 void CAlbum::MergeScrapedAlbum(const CAlbum& source, bool override /* = true */)
 {
   /*
-   We don't merge musicbrainz album ID so that a refresh of album information
-   allows a lookup based on name rather than directly (re)using musicbrainz.
-   In future, we may wish to be able to override lookup by musicbrainz so
-   this might be dropped.
-   */
-//  strMusicBrainzAlbumID = source.strMusicBrainzAlbumID;
+   Initial scraping of album information when there is a Musicbrainz album ID derived from 
+   tags is done directly using that ID, otherwise the lookup is based on album and artist names
+   but this can sometimes mis-identify the album (i.e. classical music has many "Symphony No. 5").
+   It is useful to store the scraped mbid, but we need to be able to correct any mistakes. Hence 
+   a manual refresh of album information uses either the mbid as derived from tags or the album 
+   and artist names, not any previously scraped mbid.
+  
+  A Musicbrainz album ID derived from music file tags is always taken as accurate and so can
+  not be overwritten by a scraped value. When the album does not already have an mbid or has
+  a previously scraped mbid, merge the new scraped value, flagging it as being from the
+  scraper rather than derived from music file tags.
+  */
+  if (!source.strMusicBrainzAlbumID.empty() && (strMusicBrainzAlbumID.empty() || bScrapedMBID))
+  {
+    strMusicBrainzAlbumID = source.strMusicBrainzAlbumID;
+    bScrapedMBID = true;
+  }
+
+  /*
+  Scraping can return different album artists from the originals derived from tags. When overwritting
+  the data derived from tags the original album artists are replaced with those scraped. However even
+  when not overwritting derived data, for any matching artist names the Musicbrainz artist id returned 
+  by the scraper can be used to populate any previously missing Musicbrainz artist id values.
+  */
+  if (override)
+    artistCredits = source.artistCredits; // Stores artist mbid returned by scraper    
+  else
+  {
+    // Compare original album artists with those scraped (ignoring order), and set any missing mbid
+    int matchcount = 0;
+    for (auto &artistCredit : artistCredits)
+    {
+      if (artistCredit.GetMusicBrainzArtistID().empty())
+      {
+        bool found = false;
+        auto sourceartistCredit = source.artistCredits.begin();
+        while (sourceartistCredit != source.artistCredits.end() && !found)
```

Comment:
Yep it's preferred and as @Rechi said more readable. And drop found just if string is equal set and break.